File: Soluciones/Mejor Egresado proyecto/mejorEgresado.py
num_estudiantes = int(input("Cuantos estudiantes desea ingresar: "))
prom_mayor = 0
nombres = "" #Ingrese la cantidad de estudiantes

for cont in range(num_estudiantes): #Se utiliza ciclo repetitivo for
    nombre = input("Ingrese el nombre del estudiante: ")
    promedio = float(input("Ingrese el promedio final del estudiante: "))
    
    if promedio == prom_mayor:
        nombres += " y " + nombre #Si hay más de un estudiante con el promedio más alto
    elif promedio > prom_mayor:
        prom_mayor = promedio #Se asigna el mayor promedio
        nombres = nombre

print("--------------------------------------------------------------------------------------------------")
print(f"FRUTO DE SU GRAN ESFUERZO Y DEDICACION SE LE OTORGA UN RECONOCIMIENTO A: {nombres}. POR HABER CONSEGUIDO EL PROMEDIO MAS ALTO, SIENDO ESTE DE: {prom_mayor}")
'''
Cuantos estudiantes desea ingresar: 
5
Ingrese el nombre del estudiante: 
Juan
Ingrese el promedio final del estudiante: 
7
Ingrese el nombre del estudiante: 
Carlos
Ingrese el promedio final del estudiante: 
9
Ingrese el nombre del estudiante: 
Jorge
Ingrese el promedio final del estudiante: 
8
Ingrese el nombre del estudiante: 
Daniel
Ingrese el promedio final del estudiante: 
7
Ingrese el nombre del estudiante: 
Pedro
Ingrese el promedio final del estudiante: 
9
--------------------------------------------------------------------------------------------------
FRUTO DE SU GRAN ESFUERZO Y DEDICACION SE LE OTORGA UN RECONOCIMIENTO A: Carlos y Pedro. POR HABER CONSEGUIDO EL PROMEDIO MAS ALTO, SIENDO ESTE DE: 9.0


** Process exited - Return Code: 0 **
Press Enter to exit terminal
'''

File: Soluciones/Mejor Egresado proyecto/MejorEgresado/src/mejoregresado/MejorEgresado.java
package mejoregresado;

import java.util.Scanner;

public class MejorEgresado {

    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        int numEstudiantes, cont;
        double promedio, promMayor = 0;
        String nombres = null, nombre;//Iniciaializamos las variables string en null que es el equivalente de 0 en string
        System.out.println("Cuantos estudiantes desea ingresar:");//Ingrese la cantidad de estudiantes
        numEstudiantes = teclado.nextInt();
        for (cont = 0; cont < numEstudiantes; cont++) {//Se utiliza ciclo repetitivo for
            System.out.println("Ingrese el nombre del estudiante: ");
            nombre = teclado.next();
            System.out.println("Ingrese el promedio final del estudiante: ");
            promedio = teclado.nextDouble();
            if (promedio == promMayor) {
                nombres += " y " + nombre; // Si hay más de un estudiante con el promedio más alto
            } else if (promedio > promMayor) {
                promMayor = promedio; // Se asigna el mayor promedio
                nombres = nombre;
            }
        }
        System.out.println("--------------------------------------------------------------------------------------------------");
        System.out.println("FRUTO DE SU GRAN ESFUERZO Y DEDICACION SE LE OTORGA UN RECONOCIMIENTO A: " + nombres + ". POR HABER CONSEGUIDO EL PROMEDIO MAS ALTO, SIENDO ESTE DE: " + promMayor);
    }
}
/*Cuantos estudiantes desea ingresar:
5
Ingrese el nombre del estudiante: 
Juan
Ingrese el promedio final del estudiante: 
7
Ingrese el nombre del estudiante: 
Carlos
Ingrese el promedio final del estudiante: 
9
Ingrese el nombre del estudiante: 
Jorge
Ingrese el promedio final del estudiante: 
8
Ingrese el nombre del estudiante: 
Daniel
Ingrese el promedio final del estudiante: 
7
Ingrese el nombre del estudiante: 
Pedro
Ingrese el promedio final del estudiante: 
9
--------------------------------------------------------------------------------------------------
FRUTO DE SU GRAN ESFUERZO Y DEDICACION SE LE OTORGA UN RECONOCIMIENTO A: Carlos y Pedro. POR HABER CONSEGUIDO EL PROMEDIO MAS ALTO, SIENDO ESTE DE: 9.0
BUILD SUCCESSFUL (total time: 35 seconds)
*/